### PR TITLE
docs(angular): config provider usage and content updates

### PR DESCRIPTION
**Pull request by sean-perkins**
Revamps the documentation surrounding the `Config` provider and usage in Angular. If we agree with the general layout and content adjustments, I can make similar changes to React and Vue.

Related issues: #2580

Quick preview: https://ionic-docs-git-docs-angular-config-ionic1.vercel.app/docs/angular/config

### CODE REVIEW

**Review by brandyscarney — CHANGES_REQUESTED** [review]
I'm not crazy about the formatting for the API reference & Interface, but the content additions look good to me.

File: docs/angular/config.md [hunk 1]
```diff
@@ -14,42 +14,39 @@ Ionic Config provides a way to change the properties of components globally acro
...
   ...
 })
 ```
 
-In the above example, we are disabling the Material Design ripple effect across the app, as well as forcing the mode to be Material Design.
-
 ## Per-Component Config
 
-Ionic Config is not reactive, so it is recommended to use a component's properties when you want to override its default behavior rather than set its config globally.
+Ionic Config is not reactive. It is recommended to use a component's properties or a directive when you want to override its default behavior.
```
[the remark this answers is not in the document]
Comment (sean-perkins) [reply docs/angular/config.md]: I agree this is confusing. I updated the content/example. Let me know if that helps or if there is additional changes required.

**Comment by brandyscarney** [comment]
Dropping a link to an API reference Ben sent in Slack for future reference: https://mui.com/material-ui/customization/theming/#api

File: docs/angular/config.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (brandyscarney) [reply docs/angular/config.md]: It's definitely easier to understand now, especially with the full code example for the backButtonText. 👍

File: docs/angular/config.md [hunk 2]
```diff
@@ -138,17 +147,79 @@ const getConfig = () => {
...
+  }
+}
+```
+
+## Interfaces
+
+### IonicConfig
 
-Below is a list of config options that Ionic uses.
+Below is the config options that Ionic uses.
```
Comment (brandyscarney) [inline docs/angular/config.md:222]:
```suggestion
Below are the config options that Ionic uses.
```

**Review by brandyscarney — APPROVED** [review]

**Review by amandaejohnston — APPROVED** [review]

**Pull request merged**